File: tools/marketplace-build/test_plugin.go
package main

import (
	"fmt"
	"os"
	"path/filepath"

	"github.com/spf13/cobra"
)

var testPluginCmd = &cobra.Command{
	Use:   "test-plugin [plugin-name]",
	Short: "Run tests for a single plugin",
	Args:  cobra.ExactArgs(1),
	RunE:  runTestPlugin,
}

func init() {
	rootCmd.AddCommand(testPluginCmd)
}

func runTestPlugin(cmd *cobra.Command, args []string) error {
	cmd.SilenceUsage = true
	pluginName := args[0]

	repoRoot := getRepoRoot()
	pluginPath := filepath.Join(repoRoot, "plugins", pluginName)

	// Verify plugin exists
	if _, err := os.Stat(pluginPath); os.IsNotExist(err) {
		return fmt.Errorf("plugin not found: %s", pluginPath)
	}

	// Go plugins are built and tested by the wow-look-at-my/go-toolchain action in CI.
	fmt.Printf("Testing %s\n", pluginName)
	fmt.Printf("  (no additional tests to run)\n")
	return nil
}

File: .github/scripts/test-plugin.js
#!/usr/bin/env node
const { execSync } = require("child_process");
const fs = require("fs");
const path = require("path");

const pluginName = process.argv[2];
if (!pluginName) {
  console.error("Usage: test-plugin <plugin-name>");
  process.exit(1);
}

const repoRoot = execSync("git rev-parse --show-toplevel", {
  encoding: "utf8",
}).trim();
const pluginPath = path.join(repoRoot, "plugins", pluginName);

if (!fs.existsSync(pluginPath)) {
  console.error(`plugin not found: ${pluginPath}`);
  process.exit(1);
}

console.log(`Testing ${pluginName}`);
console.log("  (no additional tests to run)");
